### PR TITLE
Add Drupal Console support

### PR DESCRIPTION
**Pull request by ben-hamelin**
## Summary

Adds Drupal Console via composer. Included in the composer json file are specific symfony libraries that needed to be downgraded in order for console to install. Specifically symfony/debug, finder, filesystem.
The compatibility issues have been raised by others and can be explored here:
https://github.com/hechoendrupal/drupal-console-core/issues/338
https://github.com/hechoendrupal/drupal-console-core/pull/347

To review the potential impacts of downgrading these 3 libraries I ran `prohibits` commands to see which packages were affected, such as:
`lando composer prohibits symfony/filesystem 3.0`
All results listed similar version requirements like '3.4|4.0' - so I think we're good.

I marked this as a work in progress as I would like to get affirmation on the approach before I finish this up, which would basically be the updated documentation on usage.

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | no (WIP)
| `CHANGELOG` reflects changes? | no (WIP)
| Unit/Functional tests reflect changes? | n/a
| Did you perform browser testing? | n/a
| Risk level | Low
| Relevant links | JIRA issue, bug reports, security alerts, etc.
https://github.com/oomphinc/drupal-scaffold/issues/45

### CODE REVIEW

**Comment by nathandentzau** [comment]
For some context, we decided not to include `drupal/console` in the scaffold due to dependency locking issues when upgrading versions of Drupal. Drupal console has dependencies on the same Symfony packages as Drupal. This decision was ~2 years ago, so I'm not sure the state of dependency management in Drupal console now but something I just wanted to call out for long term maintainability of projects using this scaffold.

The question we came across a couple of years ago is, does every project _require_ Drupal console? At the time, the answer was no, so we thought to add Drupal console on a per project basis was better than putting this on all future projects and making it harder for ourselves to handle updates.

File: composer.json [hunk 1]
```diff
@@ -44,9 +44,13 @@
         "drupal/view_unpublished": "1.x-dev",
         "drush/drush": "^9.2",
         "oomphinc/composer-installers-extender": "^1.1",
+        "symfony/debug": "^3.3.3",
+        "symfony/filesystem": "^3.4",
+        "symfony/finder": "^3.4",
         "zaporylie/composer-drupal-optimizations": "^1.0"
     },
     "require-dev": {
+        "drupal/console": "^1.9",
```
Comment (nathandentzau) [inline composer.json]: I'm also torn if this should be a dev dependency as Drush is not. :thinking:

File: composer.json [hunk 2]
```diff
@@ -44,9 +44,13 @@
         "drupal/view_unpublished": "1.x-dev",
         "drush/drush": "^9.2",
         "oomphinc/composer-installers-extender": "^1.1",
+        "symfony/debug": "^3.3.3",
```
Comment (nathandentzau) [inline composer.json]: These version constraints probably won't be an issue for new projects if `drupal/console` wasn't a dev dependency as the version requirements would be gathered in the `create-project` step. You'd normally run into these dependency issues if you were adding `drupal/console` to an existing project. IMO, this scaffold should always be treated that new features are for _new_ projects and not existing ones as that's what we've only supported.

File: composer.json (hunk 1 above)
Comment (ben-hamelin) [reply composer.json]: I could see moving drush there - it's available on Acquia and I'm guessing most servers we would deploy to. If not you could always move it into require for that specific project. Alternatively if console was moved it shouldn't have any negative impact on the server environment short of space.

File: composer.json (hunk 2 above)
Comment (ben-hamelin) [reply composer.json]: Yep you're right. Clearing out the lock file and simply running install with console listed determined the proper versions. Thanks!

**Comment by ben-hamelin** [comment]
@raubin added a new label for WIP. Just need to complete the CHANGELOG to wrap this up.

File: docs/release-workflow.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+- Fixed a bug that did xyz
+```
+
+Finally, push it up to GitHub:
+
+```
+git push --follow-tags
+```
+
+[Oomph Handbook]: http://handbook.oomphcloud.com/version-control.html#branching-strategy
```
Comment (raubin) [inline docs/release-workflow.md]: This should be removed for a public repo.

**Pull request merged**